src/gt4py/next/type_system/type_specifications.py

```diff
@@ -53,6 +53,20 @@ def __str__(self) -> str:
         return str(self.dim)
 
 
+class IndexType(TypeSpec):
+    """
+    Represents the type of an index into a dimension.
+
+    Note: Once we refactor the `Dimension` to be a Python type, the need
+    to distinguish between `DimensionType` and `IndexType` will go away.
```

Comment:
```suggestion
```
I would remove this as it is consistent with the current code base and if it is a good idea it will come up again.